src/messaging/ExchangeMgr.cpp

```diff
@@ -212,7 +210,8 @@ CHIP_ERROR ExchangeManager::UnregisterUMH(Protocols::Id protocolId, int16_t msgT
 }
 
 void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const PayloadHeader & payloadHeader,
-                                        SecureSessionHandle session, System::PacketBufferHandle msgBuf, SecureSessionMgr * msgLayer)
+                                        SecureSessionHandle session, const Transport::PeerAddress & source,
```

Comment:
we should only process node ID on exchange layer or above, if we keep the transport within ExchangeMgr as now, then we don't need to pass the source IP address to upper layer.

Reply:
Resolving it based on our slack conversation.
The address is needed as PASE session establishment using SoftAP provisioning needs to provide peer address.
SecureSessionHandle class could be refactored to carry this information. That will be done in a separate PR, as the scope of the refactor is greater than this PR.